src/IdentityInterface.php

```diff
@@ -51,9 +51,10 @@ public function canResult(string $action, $resource): ResultInterface;
      *
      * @param string $action The action/operation being performed.
      * @param mixed $resource The resource being operated on.
+     * @param array $options Application-specific scope options
      * @return mixed The modified resource.
      */
-    public function applyScope(string $action, $resource);
+    public function applyScope(string $action, $resource, array $options = []);
```

Comment:
This is the breaking change that I would be concerned about. Rolling a new major is likely our best option here. It is a simple upgrade but one we don't want to surprise people with.